scripts/tasks/src/swc.ts

```diff
@@ -1,27 +1,59 @@
-import { exec } from 'child_process';
-import { promisify } from 'util';
+import fs from 'fs';
+import path from 'path';
 
+import { transform } from '@swc/core';
 import type { Options as SwcOptions } from '@swc/core';
-import { logger } from 'just-scripts';
-
-const execAsync = promisify(exec);
+import glob from 'glob';
+import * as match from 'micromatch';
 
 type Options = SwcOptions & { module: { type: 'es6' | 'commonjs' | 'amd' } };
 
-function swcCli(options: Options) {
+async function swcTransform(options: Options) {
   const { outputPath, module } = options;
-  const swcCliBin = 'npx swc';
-  const sourceDirMap = {
-    es6: 'src',
-    commonjs: 'lib',
-    amd: 'lib',
-  };
-  const sourceDir = sourceDirMap[options.module.type];
-
-  const cmd = `${swcCliBin} ${sourceDir} --out-dir ${outputPath} --config module.type=${module?.type}`;
-  logger.info(`Running swc CLI: ${cmd}`);
-
-  return execAsync(cmd);
+  let sourceFiles: string[] = [];
+
+  if (module.type === 'es6') {
+    sourceFiles = glob.sync('src/**/*.{ts,tsx}');
+  }
+
+  if (module.type === 'commonjs' || module.type === 'amd') {
+    sourceFiles = glob.sync('lib/**/*.js');
+  }
+
+  const swcConfig = JSON.parse(fs.readFileSync(path.resolve(process.cwd(), '.swcrc'), 'utf-8'));
+
+  for (const fileName of sourceFiles) {
+    const srcFilePath = path.resolve(process.cwd(), fileName);
+    const isFileExcluded = match.isMatch(srcFilePath, swcConfig.exclude, { contains: true });
+
+    if (isFileExcluded) {
+      continue;
+    }
+
+    const sourceCode = fs.readFileSync(srcFilePath, 'utf-8');
+
+    const result = await transform(sourceCode, {
+      filename: fileName,
+      module: { type: module.type },
+      sourceFileName: path.basename(fileName),
+      outputPath,
+    });
+
+    const compiledFilePath = path.resolve(
+      process.cwd(),
+      fileName.replace(`${module.type === 'es6' ? 'src' : 'lib'}`, outputPath!),
+    );
+
+    //Create directory folder for new compiled file(s) to live in.
+    await fs.promises.mkdir(compiledFilePath.replace(path.basename(compiledFilePath), ''), { recursive: true });
+
+    const compiledFilePathJS = `${compiledFilePath.replace(/\.(tsx|ts)$/, '.js')}`;
```

Comment:
lets extract/hoist regexp to constant to not instantiate it N times. regexp creation tends to be expensive operation